beanie/odm/queries/find.py

```diff
@@ -807,6 +866,74 @@ def find_one(
         self.pymongo_kwargs.update(pymongo_kwargs)
         return self
 
+    @overload
+    async def update(
+        self,
+        *args: Mapping[str, Any],
+        session: Optional[ClientSession] = None,
+        bulk_writer: Optional[BulkWriter] = None,
+        response_type: UpdateResponse,
+        **pymongo_kwargs,
+    ) -> UpdateResponse:
+        """
+        Create Update with modifications query
+        and provide search criteria there
+
+        :param args: *Mapping[str,Any] - the modifications to apply.
+        :param session: Optional[ClientSession]
+        :param bulk_writer: Optional[BulkWriter]
+        :param response_type: UpdateResponse
+        :return: UpdateMany query
+        """
+        self.set_session(session)
```

Comment:
It looks unnecessary to keep the logic part inside the method marked with `@overload`. It will not be called at the runtime.